### PR TITLE
[PDI-13661] XSD Validator with HDFS Files

### PR DESCRIPTION
**Pull request by matthewtckr**
Add support for other VFS objects, by way of using the getInputStream for the XSD Schema file.

### CODE REVIEW

**Comment by mbatchelor** [comment]
Please include a test case for this

File: engine/src/org/pentaho/di/trans/steps/xsdvalidator/XsdValidator.java [hunk 1]
```diff
@@ -205,21 +202,20 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
 
           // We deal with XML file
           // Get XML File
-          File xmlfileValidator = new File( XMLFieldvalue );
+          FileObject xmlfileValidator = KettleVFS.getFileObject( XMLFieldvalue );
           if ( !xmlfileValidator.exists() ) {
```
Comment (mbatchelor) [inline engine/src/org/pentaho/di/trans/steps/xsdvalidator/XsdValidator.java]: Isn't it possible that this could throw an NPE ?

**Comment by matthewtckr** [comment]
Added a null check to short-circuit a possible NPE

**Comment by pamval** [comment]
Please rebase and reiisue if still applicable. This has some merge conflicts right now.

**Pull request closed without merging**